FifthTask Complete

kou-okada opened the pull request.

kou-okada committed:
FifthTask Complete

kousuke-irikura commented:
以下の要件を満たすよう、実装してください。
・日別の売上の総額を計算して出力してください。（１行に１日）
・売上が0円の日も出力されるようにしてください。
・出力結果に不要なスペース?が含まれているので、不要な文字が入らないように出力してください。
```
     販売日	,	売り上げ総額
2020/01/15	,	   20000円
```
↓
```
販売日,売上総額
2020/01/15,20000円
```

kou-okada committed:
修正したFifthTask.java

kou-okada committed:
Complete FifthTask.java

k-irikura commented on bin/.gitignore:
```diff
@@ -2,3 +2,4 @@
 /FourthTask.class
 /SecondTask.class
 /ThirdTask.class
+/FifthTask.class
```
`*.class` と書けば、ファイルが増えるたびに追記する必要が無いです。
あと、.gitignore はディレクトリごとに置くこともできますが、基本的にプロジェクトのTOP下の .gitignore で全体を制御する方が良いです。
以下のような記事なども参考にしてみてください。

https://qiita.com/tora470/items/d61c3fcb261e48edc996

kou-okada committed:
First Commit
Sales.java、SalesItem.javaにのファイルはpackage以外はいじらず、実装しました。
ImprovementFifth.javaが改良したファイルです。
コメントに各メソッドの詳細が書いています。確認お願いします。

kou-okada committed:
Second Commit
Gitgnore修正

kou-okada committed:
課題の6個目を完了しました。Customer.javaにCustomer.csvを読み込むクラスを書いています。

kousuke-irikura commented on src/Sales/Customer.java:
```diff
@@ -0,0 +1,26 @@
+package Sales;
+
+public class Customer {
+	private String customer;
```
この場合の変数名は、`customerName` のようにして下さい。
`customer`だと、Customerクラスのインスタンスのように見えてしまいます。
適切な変数名を付けるのは、コードを読みやすくする/誤読を避けるために、とても大切です。

kousuke-irikura commented on src/SixthTask.java:
```diff
@@ -0,0 +1,283 @@
+
+public class SixthTask {
+	public static void main(String[] args) {
+		ArrayList<Sales> salesArr = readSales("data/input/SalesList.csv");
```
ここは、以下のように書く方が良いです。
`List<Sales> salesList = readSales("data/input/SalesList.csv");`

・ArrayList で宣言するより、Listで宣言しましょう。Listで宣言しておくと他のList系クラスに置き換えた際に、修正箇所を１か所に限定することができます。

・変数名がsalesArr の場合、Listではなく配列のように見えてしまうので、xxxListのように宣言する方が良いです。

kousuke-irikura commented on src/SixthTask.java:
```diff
@@ -0,0 +1,283 @@
+public class SixthTask {
+	public static void main(String[] args) {
+		ArrayList<Sales> salesArr = readSales("data/input/SalesList.csv");
+		ArrayList<SalesItem> salesItemArr = readSalesItem("data/input/SalesItem.csv");
+		ArrayList<Customer> customerArr = readCustomer("data/input/Customer.csv");
```
SalesItemとCustomerは、Mapに持っておくと、他の処理のロジックをシンプルにできると思います。
（ループを１階層減らせる）
`Map<String, SalesItem> salesItemMap = new HashMap<>(); //Map<code, Object> の形式`

kousuke-irikura commented on src/Sales/Customer.java:
```diff
@@ -0,0 +1,26 @@
+package Sales;
```
パッケージ名の最初は小文字にするルールがあるので、Sales ではなく、sales が良いです。
また、一般的にこのようなエンティティクラス（と呼ばれるもの）のパッケージ名は、model 、または entity
と名付けるとよいです。

kousuke-irikura commented on bin/.gitignore:
```diff
@@ -1,4 +1,2 @@
-/FourthTask.class
-/SecondTask.class
-/ThirdTask.class
+/ItemObject.class
```
レビューの対象が分かりにくくなってしまうので、.gitignore を適切に設定して、不要なファイルをPRに含めないようにしましょう。
